pkg/apis/fleet/v1alpha1/types.go

```diff
@@ -248,6 +250,56 @@ type PodSecurityPolicy struct {
 	ValidationFailureAction string `json:"validationFailureAction,omitempty"`
 }
 
+// BackupConfig defines the configuration for backups.
+type BackupConfig struct {
+	// Chart defines the helm chart configuration of the backup engine.
+	// The default value is:
+	//
+	// chart:
+	//   repository: https://vmware-tanzu.github.io/helm-charts
+	//   name: velero
+	//   version: 5.0.2
+	//
+	// +optional
+	Chart *ChartConfig `json:"chart,omitempty"`
+
+	// Storage provides details on where the backup data should be stored.
+	Storage BackupStorage `json:"storage"`
+
+	// ExtraArgs provides the extra chart values for the backup engine chart.
+	// For example, use the following configuration to change the image tag or pull policy:
+	//
+	// extraArgs:
+	//   image:
+	//     repository: velero/velero
+	//     tag: v1.11.1
+	//     pullPolicy: IfNotPresent
+	//
+	// +optional
+	ExtraArgs apiextensionsv1.JSON `json:"extraArgs,omitempty"`
+}
+
+type BackupStorage struct {
+	// Location specifies where the backup data will be stored.
+	Location BackupStorageLocation `json:"location"`
+
+	// Credentials represents the reference to the secret for the object store.
+	// The secret must contain the fields `accessKeyID` and `secretAccessKey` to access the backup storage location.
+	// +required
+	Credentials ObjectStoreConfig `json:"credentials"`
```

Comment:
It is confusing to me about ObjectStoreConfig

```
type ObjectStoreConfig struct {
	// SecretName is the name of the secret that holds the object store configuration.
	// The path of object store configuration must be `objstore.yml`
	// +required
	SecretName string `json:"secretName"`
}
```

Reply:
Yes, it is confusing. I considered overwriting the ObjectStoreConfig, but I was worried it might disrupt the original functionality.

Reply:
The SecretName can be customized, but the SecretKey is fixed to objstore.yml.